tests/test_missing.py

```diff
@@ -6,63 +6,317 @@
 import tempfile
 import unittest
 
+from netCDF4 import Dataset
+
 from activestorage.active import Active
 from activestorage.config import *
 from activestorage import dummy_data as dd
 
 import utils
 
 
-def _doit(testfile):
-    """ 
-    Compare and contrast vanilla mean with actual means
-    """
-    uri = utils.write_to_storage(testfile)
-    active = Active(uri, "data", utils.get_storage_type())
+"""
+There is a high level of duplication in this test module:
+this is to avert any issues of I/O toestepping from h5netcdf
+when running with USE_S3=True; this will probably have to change
+when we start using netCDF4python with S3-enabled
+"""
+
+def test_partially_missing_data(tmp_path):
+    testfile = str(tmp_path / 'test_partially_missing_data.nc')
+    r = dd.make_partially_missing_ncdata(testfile)
+
+    # retrieve the actual numpy-ed result
+    ds = Dataset(testfile)
+    actual_data = ds["data"][:]
+    ds.close()
+    numpy_mean = np.ma.mean(actual_data[0:2, 4:6, 7:9])
+    print("Numpy masked result (mean)", numpy_mean)
+
+    # write file to storage
+    testfile = utils.write_to_storage(testfile)
+
+    # run the two Active instances: transfer data and do active storage
+    active = Active(testfile, "data", utils.get_storage_type())
     active._version = 0
     d = active[0:2, 4:6, 7:9]
+
+    # NOT numpy masked to check for correct Active behaviour
     mean_result = np.mean(d)
```

Comment:
nit: rename to unmasked_mean for readability?

Reply:
indeed! will do now :+1: